### ISSUE TITLE
qemu-guest-agent missing in umbrelos 1.1

### ISSUE THREAD

**Issue by FudiHub** [getumbrel/umbrel#1816]
I installed UmbrelOS 1.1 in a VM for upgrading from 0.5 following the [documentation](https://community.umbrel.com/t/how-to-update-from-umbrelos-0-5-to-umbrelos-1-1-on-linux-devices/16704#how-to-update-from-umbrelos-05-on-a-linux-vm-12).
But to get network connectivity within the VM, apt package qemu-guest-agent needs to be installed.
Considering this terminal message  when loggin in via ssh
```shell
- Warning ---------------------------------------------------------------------
| Terminal access is only enabled for debugging purposes. Any modifications   |
| made to the umbrelOS system will not be persisted between software updates. |
| For use-cases where you want to run custom software in a Linux environment, |
| consider using the Portainer app available in the Umbrel App Store.         |
-------------------------------------------------------------------------------
``` /
I assume any apt install is not surviving a umbrelos update.
So what is the approach here?
Why is qemu-guest-agent missing in umbrelos? Will my manual install survive a future update of umbrelos?

**Comment by trijenhout** [comment]
i can not get true the (i guess) grup  ***usb.img and the ***. img keeps a blinking curser in my qemu.... any tips?

**Comment by ToS0** [comment]
Yes, all you system changes (not your home dir) get lost on every update. I guess they want to sell their hardware device only, supporting less alternative installations

**Comment by highghlow** [comment]
It's strange that your VM software requires you to install guest additions to get network connectivity. I hope they come up with some system to persist root changes (Additional Packages in the Advanced settings, Running user-specified commands after an update. etc.)

> I guess they want to sell their hardware device only, supporting less alternative installations

If they use an incremental system of updating the root, then some user-installed package could be corrupted, bringing the entire system down. The only solution would be to use something like overlayfs, but you just get an immutable distro at this point.

**Comment by trijenhout** [comment]
i was of the opinion the qemu-guest-agent is in modern linux kernels.... for me it works fine... did have some trouble whit the UEFI settings, i needed to set 
  <os firmware="efi">
    <type arch="x86_64" machine="pc-q35-8.1">hvm</type>
    <firmware>
      <feature enabled="no" name="enrolled-keys"/>
      <feature enabled="no" name="secure-boot"/>
    </firmware>

secure-boot was the fix for me to get the installer booted... no trouble after that... (opensuse15.4 host)

**Comment by ebaschiera** [comment]
qemu-guest-agent is not required for networking to work.
qemu-guest-agent would be a nice addition anyway.

**Comment by grostim** [comment]
hi ! 
I need qemu-ga too, in order to deal smoothly with snapshots.

**Comment by Onepamopa** [comment]
Hey, what's the state of qemu-guest-agent? I'm running Umbrel in a VM, so kinda need that..

**Comment by ilium007** [comment]
+1 we need qemu-guest-agent

**Comment by ilium007** [comment]
If anyone else came here via google looking for qemu-guest-agent on Umbrel, do this:

In `/home/umbrel/umbrel/custom-hooks/pre-start` (create if required)

```
apt update && apt install qemu-guest-agent -y
systemctl start qemu-guest-agent
```

This will install and start qemu-guest-agent on each reboot.

**Comment by phaus** [comment]
I created an app for myself and an appstore to keep it up-to-date :)
https://github.com/phaus/app-store

**Comment by ilium007** [comment]
> I created an app for myself and an appstore to keep it up-to-date :)
> https://github.com/phaus/app-store

I didn't have time to go over your code before I blindly installed on my Umbrel.

**Comment by phaus** [comment]
Fair enough. 
I just did not want something, that performs a non deteministic install during startup, so an App was my solution.
The "code" is also quite small :)
https://github.com/phaus/qemu-ga-container/blob/master/Dockerfile
